src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -74,4 +79,19 @@ public EntityInteractHandler getItemHandler() {
     public ItemUseHandler onRightClick() {
         return PlayerRightClickEvent::cancel;
     }
-}
\ No newline at end of file
+
+    private void healZombieVillager(@Nonnull ZombieVillager zombieVillager, @Nonnull Player p) {
+        zombieVillager.setConversionTime(1);
+
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_15)) {
+            zombieVillager.setConversionPlayer(p);
+        }
+    }
+
+    private void healZombifiedPiglin(@Nonnull Entity zombiePiglin) {
```

Comment:
```suggestion
    private void healZombifiedPiglin(@Nonnull PigZombie zombiePiglin) {
```

Reply:
Casting here isnt necessary, it saves a few milliseconds so I dont think it matters

Reply:
I agree with Linox, changing the parameter type restricts the type of argument this can accept.
So this is definitely a desired change in terms of maintainability.

Reply:
> Casting here isnt necessary, it saves a few milliseconds so I dont think it matters
> 
> In the current context, were 100% sure its a PigZombie, but we dont have to cast it since we can use the generic Entity methods
> 
> Though if this is preferred, Ill change it 

You are already checking instanceof befıre this is called. That doesn't change a thing.